pkgs/os-specific/linux/nvidia-x11/builder.sh

```diff
@@ -2,8 +2,25 @@ if [ -e .attrs.sh ]; then source .attrs.sh; fi
 source $stdenv/setup
 
 unpackManually() {
+    compression=$($SHELL -r $src --info | sed -n 's/^.*Compression\s*:\s\(\S*\).*$/\1/p')
+    case $compression in
+        zstd)
+        extractCmd="zstd -d"
+        ;;
+        xz)
+        extractCmd="xz -d"
+        ;;
+        gzip)
+        extractCmd="gzip -d"
+        ;;
+        *)
+        echo "runfile uses unknown compression '$compression'"
+        return 1
+        ;;
+    esac
+
     skip=$(sed 's/^skip=//; t; d' $src)
-    tail -n +$skip $src | xz -d | tar xvf -
+    tail -n +$skip $src | $extractCmd | tar xvf -
```

Comment:
I'd prefer to use libarchive here, see https://github.com/NixOS/nixpkgs/pull/222834/commits/8ef6f5f411b0bf6d7329ef20e30820819c5ea0c8